The job **`x86_64-rust-for-linux`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/13635863040/job/38114418101) [(plain)](https://github.com/rust-lang/rust/commit/f537cdb637675aba89f67af5f0e6f18b073c7f1f/checks/38114418101/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
../x.py build --stage 2 library rustdoc clippy rustfmt
../x.py build --stage 0 cargo

BUILD_DIR=$(realpath ./build/x86_64-unknown-linux-gnu)
ls -la ${BUILD_DIR}/stage0/bin
ls -la ${BUILD_DIR}/stage1/bin
ls -la ${BUILD_DIR}/stage2/bin

export PATH=${PATH}:${BUILD_DIR}/stage2/bin
# Install rustup so that we can use the built toolchain easily, and also
# install bindgen in an easy way.
#curl --proto '=https' --tlsv1.2 -sSf -o rustup.sh https://sh.rustup.rs
#sh rustup.sh -y --default-toolchain none
---

# Install bindgen
"${BUILD_DIR}"/stage0/bin/cargo install \
  --version $(linux/scripts/min-tool-version.sh bindgen) \
  --root ${BUILD_DIR}/bindgen


ls -la ${BUILD_DIR}/bindgen/bin
export PATH=${PATH}:${BUILD_DIR}/bindgen/bin
# Configure Rust for Linux
cat <<EOF > linux/kernel/configs/rfl-for-rust-ci.config
# CONFIG_WERROR is not set

---
Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 0ad2507d5 Linux 6.14-rc3
error: the following required arguments were not provided:
  [CRATE[@<VER>]]...

Usage: cargo install --version <VERSION> --root <DIR> [CRATE[@<VER>]]...
For more information, try '--help'.
  local time: Mon Mar  3 17:30:37 UTC 2025
  network time: Mon, 03 Mar 2025 17:30:37 GMT
##[error]Process completed with exit code 1.
```

</details>

--- a/src/ci/docker/scripts/rfl-build.sh
+++ b/src/ci/docker/scripts/rfl-build.sh
@@ -8,16 +8,10 @@ LINUX_VERSION=v6.14-rc3
 ../x.py build --stage 2 library rustdoc clippy rustfmt
 ../x.py build --stage 0 cargo
 
-# Install rustup so that we can use the built toolchain easily, and also
-# install bindgen in an easy way.
-curl --proto '=https' --tlsv1.2 -sSf -o rustup.sh https://sh.rustup.rs
-sh rustup.sh -y --default-toolchain none
+BUILD_DIR=$(realpath ./build/x86_64-unknown-linux-gnu)
 
-source /cargo/env
-
-BUILD_DIR=$(realpath ./build)
-rustup toolchain link local "${BUILD_DIR}"/x86_64-unknown-linux-gnu/stage2
-rustup default local
+# Provide path to rustc, rustdoc, clippy-driver and rustfmt to RfL
+export PATH=${PATH}:${BUILD_DIR}/stage2/bin
 
 mkdir -p rfl
 cd rfl
@@ -33,9 +27,13 @@ git -C linux fetch --depth 1 origin ${LINUX_VERSION}
 git -C linux checkout FETCH_HEAD
 
 # Install bindgen
-"${BUILD_DIR}"/x86_64-unknown-linux-gnu/stage0/bin/cargo install \
+"${BUILD_DIR}"/stage0/bin/cargo install \
   --version $(linux/scripts/min-tool-version.sh bindgen) \
+  --root ${BUILD_DIR}/bindgen \
   bindgen-cli
+
+# Provide path to bindgen to RfL
+export PATH=${PATH}:${BUILD_DIR}/bindgen/bin
 
 # Configure Rust for Linux
 cat <<EOF > linux/kernel/configs/rfl-for-rust-ci.config

--- a/src/ci/github-actions/jobs.yml
+++ b/src/ci/github-actions/jobs.yml
@@ -271,9 +271,8 @@ auto:
 
   # Tests integration with Rust for Linux.
   # Builds stage 1 compiler and tries to compile a few RfL examples with it.
-  # FIXME: fix rustup 1.28 issue
-  #  - name: x86_64-rust-for-linux
-  #    <<: *job-linux-4c
+  - name: x86_64-rust-for-linux
+    <<: *job-linux-4c
 
   - name: x86_64-gnu
     <<: *job-linux-4c

